src/generic_core/core.ts

```diff
@@ -32,6 +32,7 @@ var bgAppPageChannel = new freedom();
 // Keep track of the current remote instance who is acting as a proxy server
 // for us.
 var remoteProxyInstance : Core.RemoteInstance = null;
+var remoteProxyConnection : Core.RemoteConnection = null;
```

Comment:
You can't have both remoteProxyInstance and remoteProxyConnection. you can't be proxying through two instances.

I think it'd make sense if copy-paste proxy was more like a special case of proxying, so instead of adding bunch of command to uProxy.Command enum, you'd add field indicating whether it's a copy-paste proxy or not.

I see why you wouldn't want to create a RemoteInstance object for copy-paste, you don't really need consent and there might be other things you don't need, but maybe you RemoteConnection can be a base class, and RemoteInstance be derived from that and then remoteProxyInstance would be either RemoteConnection or RemoteInstance based on what type of connection you have.

This is just what I notice by taking a quick glance at it. Do we have a design doc or something for this? I wasn't part of latest design discussions.